http_proxy.go

```diff
@@ -615,7 +618,9 @@ func (hp *HTTPProxy) listen() (net.Listener, error) {
 	case HTTPScheme:
 		return listener, nil
 	case HTTPSScheme, HTTP2Scheme:
-		return tls.NewListener(listener, hp.tlsConfig), nil
+		l := NewTLSListener(listener, hp.tlsConfig)
```

Comment:
How about we passed TLSServerConfig to NewTLSListener?

Reply:
I decided to leave it as it is, to have logging in the same place where we load certs. 
```go
func (hp *HTTPProxy) configureHTTPS() error {
	if hp.config.CertFile == "" && hp.config.KeyFile == "" {
		hp.log.Infof("no TLS certificate provided, using self-signed certificate")
	} else {
		hp.log.Debugf("loading TLS certificate from %s and %s", hp.config.CertFile, hp.config.KeyFile)
	}

	hp.tlsConfig = httpsTLSConfigTemplate()

	return hp.config.ConfigureTLSConfig(hp.tlsConfig)
}
```

Reply:
Eventually, I changed it.